deployability/modules/allocation/allocation.py

```diff
@@ -66,12 +66,21 @@ def __delete(cls, payload: models.DeletionPayload) -> None:
             payload (DeletionPayload): The payload containing the parameters
                                         for instance deletion.
         """
-        payload = models.DeletionPayload(**dict(payload))
+        payload = models.TrackPayload(**dict(payload))
         # Read the data from the track file.
         with open(payload.track_output, 'r') as f:
             track = models.TrackOutput(**yaml.safe_load(f))
         provider = PROVIDERS[track.provider]()
-        provider.destroy_instance(track.instance_dir, track.identifier, track.key_path)
+        deletion_payload = {}
+        deletion_payload['instance_dir'] = track.instance_dir
+        deletion_payload['identifier'] = track.identifier
+        deletion_payload['key_path'] = track.key_path
+        deletion_payload['platform'] = track.platform
+        deletion_payload['host_identifier'] = track.host_identifier
+        deletion_payload['host_instance_dir'] = track.host_instance_dir
+        deletion_payload['ssh_port'] = str(track.ssh_port)
+        deletion_payload['arch'] = track.arch
+        provider.destroy_instance(models.DeletionPayload(**deletion_payload))
```

Comment:
As the attributes are the came, couldn't this be simplified to something like this?
```suggestion
        provider.destroy_instance(models.DeletionPayload(**dict(track)))
```

And to ensure the port is always a string, you could create a validator in the Model Class to cast it, something like

```python
    @validator('ssh_port', pre=True)
    def validate_port(cls, value) -> str | None:
        if not value:
            return

        return str(value)
```